src/lv_message_efficient.cc

```diff
@@ -204,27 +185,52 @@ namespace grpc_labview
     {
         for (auto nestedMessage : _repeatedMessageValuesMap)
         {
-            auto fieldInfo = nestedMessage.second.get()->_fieldInfo;
-            auto buffer = nestedMessage.second.get()->_buffer;
-            auto numElements = nestedMessage.second.get()->_numElements;
+            auto& fieldInfo = nestedMessage.second.get()->_fieldInfo;
+            auto& buffer = nestedMessage.second.get()->_buffer;
+            auto numClusters = nestedMessage.second.get()->_numElements;
 
             auto metadata = fieldInfo._owner->FindMetadata(fieldInfo.embeddedMessageName);
             const char* lv_ptr = (this->GetLVClusterHandleSharedPtr()) + fieldInfo.clusterOffset;
             auto clusterSize = metadata->clusterSize;
             auto alignment = metadata->alignmentRequirement;
 
 
-            // shrink the array to the correct size
-            auto arraySize = numElements * clusterSize;
-            NumericArrayResize(0x08, 1, reinterpret_cast<void*>(const_cast<char*>(lv_ptr)), arraySize);
+            // Allocate an array with the correct size and alignment for the cluster.
+            auto byteSize = numClusters * clusterSize;
+            auto alignedElementSize = byteSize / alignment;
+            if (byteSize % alignment != 0)
+            {
+                alignedElementSize++;
+            }
+            auto numDimensions = 1;
+            NumericArrayResize(GetTypeCodeForSize(alignment), numDimensions, reinterpret_cast<void*>(const_cast<char*>(lv_ptr)), alignedElementSize);
             auto arrayHandle = *(LV1DArrayHandle*)lv_ptr;
-            (*arrayHandle)->cnt = numElements;
+            (*arrayHandle)->cnt = numClusters;
 
             auto _vectorDataPtr = buffer.data();
             auto _lvArrayDataPtr = (*arrayHandle)->bytes(0, alignment);
-            memcpy(_lvArrayDataPtr, _vectorDataPtr, arraySize);
+            memcpy(_lvArrayDataPtr, _vectorDataPtr, byteSize);
         }
-    }
 
+        for (auto repeatedStringValue : _repeatedStringValuesMap)
+        {
+            auto& fieldInfo = repeatedStringValue.second.get()->_fieldInfo;
+            auto& repeatedString = repeatedStringValue.second.get()->_repeatedString;
+            const char* lv_ptr = (this->GetLVClusterHandleSharedPtr()) + fieldInfo.clusterOffset;
+
+            NumericArrayResize(GetTypeCodeForSize(sizeof(char*)), 1, reinterpret_cast<void*>(const_cast<char*>(lv_ptr)), repeatedString.size());
```

Comment:
Could use 'numDimensions' here instead of '1' if you keep the variable.

Reply:
Sticking with the constant.